src/main/java/org/zalando/planb/provider/OIDCController.java

```diff
@@ -64,16 +64,21 @@
      */
     @RequestMapping(value = {"/oauth2/access_token", "/z/oauth2/access_token"}, method = RequestMethod.POST)
     @ResponseBody
-    OIDCCreateTokenResponse createToken(@RequestParam(value = "realm", required = true) String realmName,
+    OIDCCreateTokenResponse createToken(@RequestParam(value = "realm") Optional<String> realmNameParam,
                                         @RequestParam(value = "grant_type", required = true) String grantType,
                                         @RequestParam(value = "username", required = true) String username,
                                         @RequestParam(value = "password", required = true) String password,
                                         @RequestParam(value = "scope") Optional<String> scope,
-                                        @RequestHeader(name = "Authorization") Optional<String> authorization)
+                                        @RequestHeader(name = "Authorization") Optional<String> authorization,
+                                        @RequestHeader(name = "Host") Optional<String> hostHeader)
```

Comment:
* Host header is not required for HTTP 1.0, see https://www.w3.org/Protocols/HTTP/1.0/spec.html#Request-Header
* the Authorization header I will not touch right now (as it's unrelated to my change)